ios/Exponent/Versioned/Core/UniversalModules/EXNotifications/EXScopedNotificationsUtils.m

```diff
@@ -18,4 +18,37 @@ + (BOOL)shouldNotification:(UNNotification *)notification beHandledByExperience:
   return [EXScopedNotificationsUtils shouldNotificationRequest:notification.request beHandledByExperience:experienceId];
 }
 
++ (NSString *)scopedCategoryIdentifierWithId:(NSString *)categoryId forExperience:(NSString *)experienceId
+{
+  NSString *escapedExperienceId = [EXScopedNotificationsUtils escapedString:experienceId];
+  NSString *escapedCategoryId = [EXScopedNotificationsUtils escapedString:categoryId];
+  return [NSString stringWithFormat:@"%@/%@", escapedExperienceId, escapedCategoryId];
+}
+
++ (NSString *)unscopedCategoryIdentifierWithId:(NSString *)scopedCategoryId forExperience:(NSString *)experienceId
+{
+  NSString* scopingPrefix = [NSString stringWithFormat:@"%@/", [EXScopedNotificationsUtils escapedString:experienceId]];
+  NSString* unscopedEscapedCategoryId = [scopedCategoryId stringByReplacingOccurrencesOfString:scopingPrefix
+                                                                                    withString:@""];
+  return [EXScopedNotificationsUtils unescapedString:unscopedEscapedCategoryId];
+}
+
++ (NSString *)escapedString:(NSString*)string
+{
+  return [string stringByReplacingOccurrencesOfString:@"/" withString:@"\\/"];
```

Comment:
I think that even with escaping `\`, we could still run into issues. Like if a category ends with `\`, but also contains `\/` somewhere else in the identifier, how do we differentiate which is the actual delimiter?

```
i.e.
experienceID = 123\
Identifier = my\/category

ScopedID = 123\\/my\\\/category
```

> While it’s true that no matter what in this case, the # of `\` before the REAL delimiter is always even, while any other cases (I think) are certain to be odd, this still seems flaky, or at least hard to follow.

What do you think of making the delimiter `//`? This way, since we escape all other `/`, if there are more than one `/` in a row, that’s our delimiter.

Edge cases of a scope ending with `/`, or a categoryID beginning with `/` are addressed [here](https://github.com/expo/expo/pull/11651/files#diff-89812789ca3aaa75a377a3673054b54f3133139d24fab3d87563a0708cf6cd6cR57-R59). Plus- no regexing 😅